### PR TITLE
feature: add virtual threads to LoadEngine and validation tests

### LINKED ISSUE #10: Introduce Virtual Threads in Load Engine
## Description
Upgrade `LoadEngine` to use Virtual Threads for request execution.

## Tasks
- [ ] Create a Virtual Thread pool in `LoadEngine`
- [ ] Submit each HTTP request as a separate task to the pool instead of blocking the main loop
- [ ] Shut down the pool properly when the test ends or is stopped manually
- [ ] Update tests to verify RPS accuracy with the new model

### PR DESCRIPTION
**Pull request by marr97**
## What
Upgrade `LoadEngine` to use Virtual Threads for request execution. 
Each HTTP request is now submitted as a separate Virtual Thread task via `ExecutorService`, 
so the main loop is no longer blocked waiting for responses.

Also added:
- Input validation in LoadEngine constructor (null/empty URL, non-positive RPS/duration)
- Replaced busy-wait with `Thread.sleep()` to reduce CPU usage
- Tests for validation edge cases

## How to verify
./mvnw test

Closes #10

### CODE REVIEW

File: src/test/java/com/volta/HttpSenderTest.java
Comment (devcrocod) [inline src/test/java/com/volta/HttpSenderTest.java:1]: It's not very clear why the tests were removed

File: src/main/java/com/volta/engine/LoadEngine.java [hunk 1]
```diff
@@ -35,25 +37,39 @@ public void start() {
     long endTime = System.nanoTime() + (long) durationSeconds * 1_000_000_000L;
     long sendNextTime = System.nanoTime();
 
-    try (HttpSender sender = new HttpSender()) {
+    try (HttpSender sender = new HttpSender();
+        ExecutorService executor = Executors.newVirtualThreadPerTaskExecutor()) {
```
Comment (devcrocod) [inline src/main/java/com/volta/engine/LoadEngine.java:45]: The number of threads can grow without limit, and eventually no amount of memory will be enough

File: src/main/java/com/volta/engine/LoadEngine.java [hunk 2]
```diff
@@ -35,25 +37,39 @@ public void start() {
...
+
       while (running && System.nanoTime() < endTime) {
 
-        while (System.nanoTime() < sendNextTime) {
-          // busy-wait
-        }
+        long waitMillis = (sendNextTime - System.nanoTime()) / 1_000_000;
```
Comment (devcrocod) [inline src/main/java/com/volta/engine/LoadEngine.java:49]: With this calculation, `waitMillis` may become 0, which will cause the sleep to be skipped and the requests to be sent immediately

File: src/main/java/com/volta/engine/LoadEngine.java [hunk 3]
```diff
@@ -35,25 +37,39 @@ public void start() {
...
+              }
+            });
+
         sendNextTime += intervalNanos;
       }
     } catch (Exception e) {
       log.error("Sender closed or failed to initialize", e);
     }
 
+    running = false;
```
Comment (devcrocod) [inline src/main/java/com/volta/engine/LoadEngine.java:90]: It’s not obvious how the entire process terminates

**Review by devcrocod — APPROVED** [review]
